src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1488,6 +1489,161 @@ void addIfNotNull(INamedTypeSymbol? toAdd)
+                    return;
+
+                // Delegates have a synthesized operator for combining them together.
+                if (type.TypeKind == TypeKind.Delegate)
+                    return;
+
+                switch (type.OriginalDefinition.SpecialType)
+                {
+                    case SpecialType.System_SByte:
+                    case SpecialType.System_Byte:
+                    case SpecialType.System_Int16:
+                    case SpecialType.System_UInt16:
+                    case SpecialType.System_Int32:
+                    case SpecialType.System_UInt32:
+                    case SpecialType.System_Int64:
+                    case SpecialType.System_UInt64:
+                    case SpecialType.System_IntPtr:
+                    case SpecialType.System_UIntPtr:
+                    case SpecialType.System_Char:
+                    case SpecialType.System_Single:
+                    case SpecialType.System_Double:
+                    case SpecialType.System_Decimal:
+                    case SpecialType.System_Boolean:
+                    case SpecialType.System_Object:
+                    case SpecialType.System_String:
+                    case SpecialType.System_DateTime:
+                    case SpecialType.System_Delegate:
+                        return;
+                }
+
+                // ReadOnlySpan<byte> supports a builtin + operator.
+                if (type.OriginalDefinition.Equals(this.CommonGetWellKnownType(WellKnownType.System_ReadOnlySpan_T)?.GetISymbol(), SymbolEqualityComparer.ConsiderEverything) &&
+                    ((INamedTypeSymbol)type).TypeArguments[0].SpecialType == SpecialType.System_Byte)
+                {
+                    return;
+                }
+
+                throw new ArgumentException($"Unsupported built-in operator type: {type.ToDisplayString()}", paramName);
+            }
+        }
+
+        protected abstract IMethodSymbol CommonCreateBuiltinOperator(string name, ITypeSymbol returnType, ITypeSymbol leftType, ITypeSymbol rightType, bool isChecked);
+
+        /// <summary>
+        /// Creates an <see cref="IMethodSymbol"/> whose <see cref="IMethodSymbol.MethodKind"/> is <see
+        /// cref="MethodKind.BuiltinOperator"/> for a unary operator. Built-in operators are commonly created for
+        /// symbols like <c>bool int.operator==(int v1, int v2)</c> which the language implicitly supports, even if such
+        /// a symbol is not explicitly defined for that type in either source or metadata.
+        /// </summary>
+        /// <param name="name">The unary operator name.  Should be one of the names from <see cref="WellKnownMemberNames"/>.</param>
+        /// <param name="returnType">The return type of the unary operator.</param>
+        /// <param name="valueType">The type the operator applies to.</param>
+        /// <param name="isChecked">Whether or not this is a unary binary operator..</param>
```

Comment:
>unary binary operator.. [](http://example.com/codeflow?start=61&length=23)

This looks like a typo, including double dot at the end.

Reply:
:D

Reply:
Fixed.